File: siserp-logistica-backend/src/main/java/com/besoft/siserp/lgtc/dto/request/InventarioModificarCompraDetRequest.java
package com.besoft.siserp.lgtc.dto.request;

import java.io.Serializable;
import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class InventarioModificarCompraDetRequest implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long idDetCompraEstante;
	private Long idEstante;
	private Long idInventario;
	private Long idProducto;
	private Double cantidad;
	private Double precio;
	private List<InventarioCodigoBarraDetModificarRequest> listaCodigoBarra;
	private List<InventarioCodigoBarraDetModificarRequest> listaCodigoBarraElim;

}

File: siserp-logistica-frontend/src/main/siserp-logistica-frontend/src/app/modules/intranet/dto/request/inventario-modificar-compra-det.request.ts
import { InventarioCodigoBarraDetModificarRequest } from "./inventario-codigo-barra-det-modificar.request";

export class InventarioModificarCompraDetRequest {
    idDetCompraEstante: number;
    idEstante: number;
    idInventario: number;
    idProducto: number;
    cantidad: number;
    precio: number;
    listaCodigoBarra: InventarioCodigoBarraDetModificarRequest[];
    listaCodigoBarraElim: InventarioCodigoBarraDetModificarRequest[];

    nomProducto: string;
    precioCompra: number;
    ganancia: number;
    idAlmacen: number;
    nomAlmacen: string;
    nomEstante: string;

    precioAnterior: number;
}